src/Beacon.jsx

```diff
@@ -1,111 +1,204 @@
-import React, { PropTypes } from 'react';
-import Portal from 'react-portal';
-import Tooltip from 'react-portal-tooltip';
-const withClickOutside = require('react-onclickoutside');
+import React, { Children, PropTypes } from 'react';
+import ReactDOM from 'react-dom';
+import sha1 from 'sha1';
+import withClickOutside from 'react-onclickoutside';
+import Config from './BeaconConfig';
+import TetherComponent from 'react-tether';
 
-import '../assets/style.less';
+import '../assets/style.styl';
 
-const BEACON_HEIGHT = 30;
-const BEACON_WIDTH = 30;
+const TOOLTIP_OVERLAY_CLASS = 'tour-overlay';
+const TOOLTIP_FADED_CLASS = 'tour-faded';
+const TARGET_CLONE_ID = 'tour-target-clone';
+// If margins are within tolerance then we center tooltip
+const TOOLTIP_TOLERANCE = 50;
 
-export default withClickOutside(class Beacon extends React.Component {
+export class Beacon extends React.Component {
 
   static propTypes = {
-    parent: PropTypes.string,
-    position: PropTypes.string,
-    children: PropTypes.node
+    persistent: PropTypes.bool,
+    tooltipText: PropTypes.string.isRequired,
```

Comment:
You mean have the tooltip text be inside a `span` or something? I guess I could have two children (text and actual child) like `TetherComponent` does.